nan.h

```diff
@@ -801,44 +801,62 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
+        v8::Isolate::GetCurrent()->GetCurrentContext()->Global() :
+            target->ToObject();
     return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+        v8::Isolate::GetCurrent(), target_, func, argc, argv));
 #else
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
+        v8::Isolate::GetCurrent()->GetCurrentContext()->Global() :
+            Nan::To<v8::Object>(target).ToLocalChecked();
     return node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv);
+        v8::Isolate::GetCurrent(), target_, func, argc, argv);
 #endif
   }
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::String> symbol
     , int argc
     , v8::Local<v8::Value>* argv) {
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
```

Comment:
I remember now: node.js looks up the object's context with `recv->CreationContext()`.  That doesn't work when recv is a Value.